PWGHF/TableProducer/candidateCreator2Prong.cxx

```diff
@@ -728,13 +737,45 @@ struct HfCandidateCreator2ProngExpressions {
 
     // Match generated particles.
     for (const auto& particle : mcParticles) {
+      // Reject particles from background events
+      if (particle.fromBackgroundEvent() && rejectBackground) {
+        continue;
+      }
+
       flag = 0;
       origin = 0;
       std::vector<int> idxBhadMothers{};
 
+      // Slice the collisions table to get the collision info for the current MC collision
       auto mcCollision = particle.mcCollision();
-
-      const auto rejectionMask = hfEvSelMc.getHfMcCollisionRejectionMask<BCsInfo>(mcCollision);
+      auto collSlice = collInfos.sliceBy(colPerMcCollision, mcCollision.globalIndex());
+      float centrality{-1.f};
+      float mult{-1.f};
+      // Loop over the collisions in the current MC collision, and select the one with the highest FT0C multiplicity
+      for (const auto& collision : collSlice) {
+        float multColl{-1.f};
+        float centColl{-1.f};
+        switch (centralityEstimator) {
+          case 0: // FT0C
+            multColl = collision.multFT0C();
+            centColl = collision.centFT0C();
+            break;
+          case 1: // FT0M
+            multColl = collision.multFT0M();
+            centColl = collision.centFT0M();
+            break;
+          default:
+            LOG(info) << "Unknown centrality estimator, fallback to FT0C";
+            multColl = collision.multFT0C();
+            centColl = collision.centFT0C();
+            break;
+        }
+        if (mult < multColl) {
+          mult = multColl;
+          centrality = centColl;
+        }
+      } // end loop over collisions
+      const auto rejectionMask = hfEvSelMc.getHfMcCollisionRejectionMask<BCsInfo>(mcCollision, centrality);
```

Comment:
As for the reconstructed counterpart, I would pass the centrality by reference to be assigned inside the method